1_fetch/src/fetch_wqp_data.R

```diff
@@ -0,0 +1,45 @@
+fetch_wqp_data <- function(wqp_inventory, characteristics, query_site_limit = 500, 
+                           min_results = 1, min_date = "", wqp_args = NULL){
+  
+  siteids <- wqp_inventory %>%
+    pull(MonitoringLocationIdentifier) %>%
+    unique()
+  n_total_sites <- length(siteids) 
+  
+  # Break up sites into multiple WQP calls to avoid timeout issues 
+  start_time <- Sys.time()
+  sites_chunk <- ceiling(n_total_sites/query_site_limit)
+
+  # Pull data for each chunk
+  wqp_data_ls <- list()
+  
+  for(i in seq_len(sites_chunk)){
```

Comment:
I think the for loop within the target is a bit brittle for data downloads. We should take advantage of the `mapping` capabilities!